app/main/lib/shared_models/shared_model.py

```diff
@@ -30,33 +30,46 @@ def import_model_class(model_class):
     return getattr(importlib.import_module('app.main.lib.shared_models.%s' % class_name), model_class)
 
   @staticmethod
-  def start_server(model_class, model_key, options={}):
-    class_ = SharedModel.import_model_class(model_class)
+  def get_server(model_key, options={}):
+    class_ = SharedModel.import_model_class(SharedModel.get_model_class(model_key))
     instance = class_(model_key, options)
     instance.load()
-    app.logger.info('[%s] Serving model...', model_key)
+    return instance
+
+  @staticmethod
+  def register_server(model_class, model_key, options={}):
     r = redis_client.get_client()
     r.set('SharedModel:%s' % model_key, json.dumps({
       'model_class': model_class,
       'model_key': model_key,
       'options': options
     }))
     r.sadd('SharedModel', model_key)
+
+  @staticmethod
+  def start_server(model_class, model_key, options={}):
+    app.logger.info('[%s] Serving model...', model_key)
+    SharedModel.register_server(model_class, model_key, options)
+    instance = SharedModel.get_server(model_key, options)
     instance.bulk_run()
 
+  @staticmethod
+  def get_model_info(model_key):
+    r = redis_client.get_client()
```

Comment:
if this is a static method, where is it getting the redis_client instance?

Reply:
Looks like `redis_client` is a global variable imported on line 12 `from app.main.lib import redis_client`

Reply:
oh!  does that mean we are instantiating new Redis clients with every request then?

Reply:
@skyemeedan yes we are instantiating new Redis clients with every request - the latency for doing this is pretty minimal, and I've observed in other contexts before that holding them open for a while can actually induce problems. If you look at `check-api` we actually also use the same pattern of instantiating on use rather than having some abstracted session handler logic or a permanent open connection. I'm happy to change this if anyone wants in a separate ticket!

Reply:
clearly it has been working, but I'm wondering if that might be causing memory bloat?

Reply:
I dont _think so_? After execution of this function, `r` would no longer be used and hence would be eligible for garbage collection. I don't think you _need_ to actually close a connection to redis either - or at least I haven't ever...